kannon/master.py

```diff
@@ -115,69 +100,50 @@ def _exec_gokart_task(self, task: gokart.TaskOnKart) -> None:
 
     def _exec_bullet_task(self, task: TaskOnBullet) -> None:
         # Run on child job
         serialized_task = gokart.TaskInstanceParameter().serialize(task)
         job_name = gen_job_name(f"{self.job_prefix}-{task.get_task_family()}")
-        job = self._create_job_object(
+        job = self._create_child_job_object(
             job_name=job_name,
             serialized_task=serialized_task,
         )
         create_job(self.api_instance, job, self.namespace)
-        logger.info(
-            f"Created child job {job_name} with task {self._gen_task_info(task)}"
-        )
+        logger.info(f"Created child job {job_name} with task {self._gen_task_info(task)}")
         task_unique_id = task.make_unique_id()
         self.task_id_to_job_name[task_unique_id] = job_name
 
-    @staticmethod
-    def _gen_task_info(task: gokart.TaskOnKart) -> str:
-        return f"{task.get_task_family()}_{task.make_unique_id()}"
-
-    def _create_job_object(self, serialized_task: str, job_name: str) -> client.V1Job:
+    def _create_child_job_object(self, job_name: str, serialized_task: str) -> client.V1Job:
         # TODO: use python -c to avoid dependency to execute_task.py
         cmd = [
             "python",
             self.path_child_script,
             "--serialized-task",
             f"'{serialized_task}'",
         ]
+        job = deepcopy(self.template_job)
+        # replace command
+        assert len(job.spec.template.spec.containers[0].command) == 0, \
+            "command will be replaced by kannon, so you shouldn't set any command and args"
+        job.spec.template.spec.containers[0].command = cmd
+        # replace env
         child_envs = []
         for env_name in self.env_to_inherit:
             if env_name not in os.environ:
                 raise ValueError(f"Envvar {env_name} does not exist.")
             child_envs.append({"name": env_name, "value": os.environ.get(env_name)})
-        container = client.V1Container(
-            name=self.container_name,
-            image=self.image_name,
-            command=cmd,
-            env=child_envs,
-        )
-        template = client.V1PodTemplateSpec(
-            metadata=client.V1ObjectMeta(labels={"app": "kannon"}),
-            spec=client.V1PodSpec(
-                restart_policy="Never",
-                containers=[container],
-                service_account_name=self.service_account_name,
-            ),
-        )
-        spec = client.V1JobSpec(template=template, backoff_limit=self.backoff_limit)
-        job = client.V1Job(
-            api_version="batch/v1",
-            kind="Job",
-            metadata=client.V1ObjectMeta(
-                name=job_name,
-                namespace=self.namespace,
-            ),
-            spec=spec,
-        )
+        job.spec.template.spec.containers[0].env.extend(child_envs)
+        # replace job name
+        job.metadata.name = job_name
```

Comment:
assert here too as well as `command` ?

Reply:
`job.metadata.name` is an essential field, so following this rule users will set some temporary name like `child-job-template`. 
So, the solution I came up with is:
"Writing docs on what is replaced when creating child job from child job template"

I think the assertion here for `job_name` should not be added.